include/boost/math/fft/gsl_backend.hpp

```diff
@@ -174,6 +174,34 @@ namespace fft { namespace detail {
       gsl_fft_halfcomplex_wavetable *halfcomplex_wtable;
       gsl_fft_real_workspace        *real_wspace;
       
+      void pack_halfcomplex(real_value_type* out) const
```

Comment:
The algorithm that we use for rfft for powers of 2, comes naturally in-place if we use the halfcomplex representation used by FFTW. Hence we have to pack and unpack the halfcomplex returned by GSL to make it compatible with our representation.

Reply:
Wow, I appreciate your comments about what's going on in the code.

It's a bit unrelated to this comment, but I noticed that we could replace `2*boost::math::constants::pi<RealType>()` with `boost::math::constants::two_pi<RealType>()`. This could eliminate 1 ULP of error.